Feature/absolute position

joshmarom opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/pojome/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

KingYes committed:
Merge pull request #6537 from matipojo/feature/background-upgrades

Upgrades: New - Run upgrades in background.

arielk committed:
text domain

arielk committed:
Safe mode review

arielk committed:
Safe mode links

KingYes committed:
Merge pull request #6570 from arielk/release/2.4.0

safe mode review

nuritsha committed:
widgets/video: tweak - after review

nuritsha committed:
base/background-task: fix - get_current_offset

matipojo committed:
Safe Mode: Progress.

kobizz committed:
Style fix for dialog messages

nuritsha committed:
core/logger: fix - add esc_html before printing log messages

kobizz commented on assets/dev/js/editor/elements/views/behaviors/draggable.js:
```diff
@@ -0,0 +1,142 @@
+			view = this.view,
+			viewSettingsChangedMethod = view.onSettingsChanged;
+
+		view.onSettingsChanged = function() {
```
Use arrow functions and avoid use of `self`

kobizz commented on assets/dev/scss/frontend/_global.scss:
```diff
@@ -101,6 +106,117 @@
 	}
 }
 
+
```
You should export the exist `ui-resizable` rules into a new file and import it in editor & editor preview, instead of duplicating the code.

kobizz commented on assets/dev/js/editor/elements/views/behaviors/draggable.js:
```diff
@@ -0,0 +1,142 @@
+var DraggableBehavior;
+
+DraggableBehavior = Marionette.Behavior.extend( {
```
Please use ES6 syntax along this file

nuritsha committed:
includes/db: tweak - use call_user_func

KingYes committed:
Merge pull request #6574 from matipojo/release/2.4.0

Safe Mode: Progress.

KingYes committed:
Merge pull request #6576 from nuritsha/fix/loggers-esc_html

core/logger: fix - add esc_html before printing log messages

KingYes committed:
Fix text domain && phpcs

nuritsha committed:
includes/db: tweak - use call_user_func

KingYes committed:
Merge pull request #6572 from nuritsha/tweak/finishes-video-external-url

widgets/video: tweak - external url - after review

matipojo committed:
Safe Mode: Progress.

joshmarom committed:
Popup first section handles inside

joshmarom committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

nuritsha committed:
widgets/video: tweak - use control-url for external_url

KingYes committed:
Merge pull request #6579 from nuritsha/tweak/video-control-url

widgets/video: tweak - use control-url for external_url

matipojo committed:
Safe Mode: Progress.

KingYes committed:
Merge pull request #6580 from matipojo/release/2.4.0

 Safe Mode: Progress.

joshmarom committed:
Improved solution for popup first section handles position

joshmarom committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

KingYes committed:
Merge pull request #6578 from joshmarom/release/2.4.0

 Popup first section handles inside

arielk committed:
Safe mode review

KingYes committed:
Merge pull request #6582 from arielk/release/2.4.0

Safe mode review

arielk committed:
Alpha Review

arielk committed:
Merge pull request #6586 from arielk/release/2.4.0

Alpha Review

matipojo committed:
Safe Mode: Progress.

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

nuritsha committed:
js/admin and gutenberg: new - add 'back to wp editor' confirm popup

matipojo committed:
Merge branch 'release/2.4.0'

matipojo committed:
Widgets/Read More: Change - Default Link Text.

matipojo committed:
Safe Mode: Progress.

KingYes committed:
Merge pull request #6589 from matipojo/release/2.4.0

 Safe Mode & Read More

kobizz committed:
Fixed media control aspect ratio issue

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

KingYes committed:
Merge pull request #6591 from kobizz/release/2.4.0

Release/2.4.0

matipojo committed:
Upgrades: Fix - Show admin `completed` notice only if did some task.

matipojo committed:
Upgrades: Change - Run immediately.

KingYes committed:
Merge pull request #6592 from matipojo/release/2.4.0

 Upgrades: fixes

nuritsha committed:
core/admin: tweak - change wording

joshmarom committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into feature/flex-columns

arielk committed:
Data updater - review

KingYes committed:
Merge pull request #6595 from arielk/release/2.4.0

DB updater - review

kobizz committed:
Document: Removed `groups` and `library_view` unused functionality

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

kobizz committed:
ES6 style

kobizz committed:
Make the remote library system more dynamic

matipojo committed:
Upgrades: Change - Notices texts.

matipojo committed:
Logger/DB: Fix - Truncate log.

matipojo committed:
Safe Mode: Progress.

matipojo committed:
Upgrades: Fix - Percent of progress.

kobizz committed:
Fixed & improved library screens behavior

kobizz committed:
Library: Fixed input text transform

kobizz committed:
Removed redundant BC

nuritsha committed:
core/logger: fix - esc_html in log-item constructor

matipojo committed:
Safe Mode: Progress.

KingYes committed:
Merge pull request #6597 from matipojo/release/2.4.0

2.4.0 fixes

nuritsha committed:
core/logger: fix - esc_html in log-item constructor

KingYes committed:
Merge pull request #6599 from nuritsha/fix/logger-esc_html_2

core/logger: fix - esc_html in log-item constructor

kobizz committed:
`setTemplatesPage` => `setScreen`

kobizz committed:
Templates Library: Added ability to set a default screen

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

kobizz committed:
Removed irrelevant test

KingYes committed:
Merge pull request #6598 from kobizz/release/2.4.0

Release/2.4.0

matipojo committed:
Library: Fix - Limit category column width.

KingYes committed:
Bump alpha1

matipojo committed:
Safe Mode: Progress.

matipojo committed:
Safe Mode: Progress.

kobizz committed:
Allow to show the "add new" modal by clicking the "add new" button, also when already existing on the page

KingYes committed:
Merge pull request #6601 from kobizz/release/2.4.0

Release/2.4.0

KingYes committed:
Merge pull request #6602 from matipojo/release/2.4.0

Release/2.4.0 Fixes

KingYes committed:
Merge branch 'release/2.4.0' of github.com:pojome/elementor into release/2.4.0

KingYes committed:
Saved templates: Rename tab "Content" => "Page"

KingYes committed:
Merge pull request #6588 from nuritsha/feature/back-to-wp-notice

js/admin and gutenberg: new - add 'back to wp editor' confirm popup

arielk committed:
strings

arielk committed:
Fix popup text alignment

KingYes committed:
Fix redirect

KingYes committed:
Merge pull request #6604 from arielk/release/2.4.0

fix modal alignment

kobizz committed:
`remote_library` => `remoteLibrary`

kobizz committed:
Allow document to automatically import page settings

kobizz committed:
Updated dialog library

bainternet committed:
Tweak: moved animations filter to additional animations correct #6547

bainternet committed:
Tweak: run merge only once

KingYes committed:
Merge pull request #6608 from kobizz/release/2.4.0

Release/2.4.0

bainternet committed:
use local vars

KingYes committed:
Merge pull request #6609 from bainternet/tweak/animations

Tweak: run merge only once

joshmarom committed:
Make hover border-radius responsive

KingYes committed:
Merge pull request #6613 from joshmarom/release/2.4.0

Make hover border-radius responsive

KingYes committed:
Remove Partner IDs

matipojo committed:
JS / Controls Stack: New - On activate a `section` trigger an event with current namespace.

arielk committed:
Added themes page to finder

KingYes committed:
Merge pull request #6615 from arielk/release/2.4.0

Added themes page to finder

matipojo committed:
Saver: Fix - Use the correct document config.

kobizz committed:
Elementor Frontend Modules: Added jQuery dependency

kobizz committed:
Panel Footer Menu: Don't close sub menus on publish button click

matipojo committed:
JS/Controls Stack: Fix - Rename `parentName` => `controlName`.

KingYes committed:
Merge pull request #6614 from matipojo/release/2.4.0

JS / Controls Stack: New - On activate a `section` trigger an event

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

KingYes committed:
Merge pull request #6629 from kobizz/release/2.4.0

Release/2.4.0

KingYes committed:
Changed Icon for Widget Read More

KingYes committed:
Bump 2.4.0-beta1

kobizz committed:
Return back unused method as BC

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

KingYes committed:
Merge pull request #6631 from kobizz/release/2.4.0

Release/2.4.0

kobizz committed:
Frontend: Added user config

nuritsha committed:
includes/db: tweak - pass $args to call_user_func

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

arielk committed:
Icon + text

KingYes committed:
Merge pull request #6632 from arielk/release/2.4.0

Icon + text

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

KingYes committed:
Merge pull request #6573 from nuritsha/fix/background-task

base/background-task: fix - get_current_offset

KingYes committed:
phpcs

KingYes committed:
Merge pull request #6633 from kobizz/release/2.4.0

Release/2.4.0

KingYes committed:
Check variable before used

KingYes committed:
Bump 2.4.0-beta2

kobizz committed:
Local Templates: Added new `get_admin_url` method

kobizz committed:
Fixed Finder saved templates url

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

KingYes committed:
Merge pull request #6635 from kobizz/release/2.4.0

Release/2.4.0

arielk committed:
text

KingYes committed:
Merge pull request #6637 from arielk/release/2.4.0

text

joshmarom committed:
Fix some issues with responsive group controls conditions

bainternet committed:
Fix: `device_args` conditions in group control

KingYes committed:
Merge pull request #6645 from bainternet/fix/group-deviceargs

Fix: `device_args` selectors and conditions in group control

KingYes committed:
phpcs

matipojo committed:
Connect: Fix - Submenu hook for Non-English sites.

joshmarom committed:
Final adjustments for Background control group

KingYes committed:
Merge pull request #6650 from joshmarom/release/2.4.0

Final adjustments for Background control group

KingYes committed:
Merge pull request #6649 from matipojo/release/2.4.0

Connect: Fix -Admin Submenu hook for Non-English sites.

KingYes committed:
phpcs

KingYes committed:
Merge branch 'release/2.4.0' of github.com:pojome/elementor into release/2.4.0

KingYes committed:
phpcs

kobizz committed:
Templates Library: Apply blocks style on every non-page templates

KingYes committed:
Merge pull request #6659 from kobizz/release/2.4.0

Release/2.4.0

kobizz committed:
Updated Dialog library

KingYes committed:
Bump 2.4.0 beta 3

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

kobizz committed:
Make the introduction functionality more generic

joshmarom committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into feature/flex-columns

KingYes committed:
Merge pull request #6661 from kobizz/release/2.4.0

Release/2.4.0

kobizz committed:
'Introduction' => 'Right Click Introduction'

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

matipojo committed:
Documents: Fix - On create - save the template-type immediately. (Fix #6648)

kobizz committed:
Safe Mode: Added additional check

KingYes committed:
Merge pull request #6664 from kobizz/release/2.4.0

Release/2.4.0

kobizz committed:
Safe Mode: Fixed content and style

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

KingYes committed:
Merge pull request #6673 from kobizz/release/2.4.0

Release/2.4.0

KingYes committed:
Merge pull request #6669 from matipojo/release/2.4.0

Documents: Fix - On create - save the template-type immediately.

KingYes committed:
Move the "Categories" menu to end only if exists

KingYes committed:
Merge branch 'release/2.4.0' of github.com:pojome/elementor into release/2.4.0

matipojo committed:
Safe Mode: Fix - Update allowed plugins on activate/deactivate plugins.

matipojo committed:
Safe Mode: Fix - Remove `on_add_safe_mode`. Use only `update_option`.

KingYes committed:
Merge pull request #6705 from matipojo/release/2.4.0

 Safe Mode: Fixes

arielk committed:
Readme: 30 free widgets

KingYes committed:
Merge pull request #6709 from arielk/release/2.4.0

Readme: 30 free widgets

kobizz committed:
Updated Dialog library

matipojo committed:
E2E: Fix - Avoid admin slash redirection.

matipojo committed:
E2E: Fix - Gutenberg Compatibility.

KingYes committed:
Merge pull request #6711 from kobizz/release/2.4.0

Release/2.4.0

matipojo committed:
E2E: Fix - Selector for panel saver.

matipojo committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

matipojo committed:
Safe Mode: Fix - Watch if Elementor is loaded and hide the notice.

joshmarom committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into feature/flex-columns

matipojo committed:
QUnit: Remove residues.

kobizz committed:
Number Control: Added max width

kobizz committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into release/2.4.0

arielk committed:
Fix E icon style in twentyseventeen

arielk committed:
Align icons start to fix #6507

arielk committed:
Fix #6507

KingYes committed:
Merge pull request #6717 from arielk/release/2.4.0

Fix E icon style in twentyseventeen

KingYes committed:
Merge pull request #6715 from kobizz/release/2.4.0

Release/2.4.0

KingYes committed:
Merge pull request #6712 from matipojo/release/2.4.0

 Safe Mode & E2E: Fixes

KingYes committed:
Update changelog;

KingYes committed:
Update changelog

KingYes committed:
Merge

joshmarom committed:
Merge branch 'release/2.4.0' of https://github.com/pojome/elementor into feature/flex-columns

KingYes committed:
Bump to 2.4.0

KingYes committed:
Update changelog

KingYes committed:
Update changelog

matipojo committed:
E2E: Start test Safe Mode

matipojo committed:
E2E: New - Test Safe Mode

KingYes committed:
Merge pull request #6723 from matipojo/feature/e2e

Feature/e2e

KingYes committed:
Menu Anchor: ID field allow chars: `A-Z,a-z,0-9,_,-`

KingYes committed:
Tweak: Admin menu position break other plugins

arielk committed:
Added note for Anchor Menu widget

arielk committed:
Merge remote-tracking branch 'pojome/release/2.4.1' into release/2.4.1

kobizz committed:
Fixed duplicate run of elements handlers on nested documents

bainternet committed:
Tweak: Avoid widget rendering on `box-shadow` and `text-shadow` changes closes #6737

KingYes committed:
Merge pull request #6740 from kobizz/release/2.4.1

Release/2.4.1

KingYes committed:
Merge pull request #6741 from bainternet/fix/i-6737

Tweak: Avoid widget rendering on `box-shadow` and `text-shadow` changes closes #6737

KingYes committed:
Update changelog

KingYes committed:
Merge branch 'master' of github.com:pojome/elementor

KingYes committed:
Bump to 2.4.1

KingYes committed:
Update changelog

joshmarom committed:
Progress with inline widgets

joshmarom committed:
Merge branch 'release/2.4.1' of https://github.com/pojome/elementor into feature/flex-columns

joshmarom committed:
Merge branch 'release/2.4.2' of https://github.com/pojome/elementor into feature/flex-columns

matipojo committed:
E2E: New - Test frontend handlers.

KingYes committed:
Merge pull request #6757 from matipojo/release/2.4.2

E2E: New - Test frontend handlers.

joshmarom committed:
Fixes and touch-ups after code review

joshmarom committed:
Merge branch 'feature/flex-columns' of https://github.com/joshmarom/elementor into feature/absolute-position

joshmarom committed:
Merge inline columns

matipojo committed:
Logger/PHP: Fix - Log only Elementor related errors.

KingYes committed:
Merge pull request #6762 from matipojo/release/2.4.2

Logger/PHP: Fix - Log only Elementor related errors.

kobizz committed:
Removed old BC check

kobizz committed:
Fonts Manager: Improved `get_fonts` method preformance

nuritsha committed:
core/logger: fix - truncate times_date array

nuritsha committed:
core/logger: fix - truncate times_date array

KingYes committed:
Merge pull request #6770 from nuritsha/fix/logger-times-date

core/logger: fix - truncate times_date array

matipojo committed:
Background Task: Fix - Fix Total number.

ramiy committed:
i18n: merge similar translation strings

Merge two similar translation strings. And hard-code the `360px` value, this way translators won't  be able enter wrong value by mistake.

ramiy committed:
i18n: simplify translations string

Make it easier to translate the Translation String, by removing the HTML tags out of the string into a separate line.

ramiy committed:
i18n: split translation strings

Make it easier to translate the Translation String, by splitting it into two smaller strings. The second string already exist.

ramiy committed:
i18n: avoid using HTML tags in translation strings

Not all the translators know HTML, this is why you should avoid using unneeded HTML tags where they are not needed.

Replace: `<strong>%s</strong>` with simple `%s`, moving the HTML tags out of the string.

KingYes committed:
Merge pull request #6772 from matipojo/release/2.4.2

Background Task: Fix - Fix Total number.

KingYes committed:
Merge pull request #6776 from ramiy/patch-5

i18n: avoid using HTML tags in translation strings

KingYes committed:
Merge pull request #6774 from ramiy/patch-3

i18n: simplify translations string

KingYes committed:
Merge pull request #6773 from ramiy/patch-2

i18n: merge similar translation strings

KingYes committed:
Merge pull request #6775 from ramiy/patch-4

i18n: split translation strings

ramiy committed:
Bug Fix: styling image widget captions

This PR fixes the wrong condition used when displaying the caption styling section.

Currently the "**Caption**" section under the "**Style**" tab displayed only when `"caption_source" = "custom"` and when `"caption_source" != ""`.

The user can't really style the caption when he choose to use "Attachment Caption" because of this wrong condition set to display the section.

ramiy committed:
i18n: use %s placeholder in translation strings

Don't allow translators to control those chars! Hard-code them. This way no-one can accidentally change them.

ramiy committed:
i18n: use %s placeholder in translation strings

Don't allow translators to change the action name! Hard-code it. This way no-one can accidentally change the action.

ramiy committed:
i18n: use %s placeholder in translation strings

Don't allow translators to change the function name! Hard-code it. This way no-one can accidentally change the function.

kobizz committed:
Inline Editing: Increase pen menu z-index

kobizz committed:
Revisions: Load revisions by ajax, in order to decrease editor loading time

kobizz committed:
Revisions: Fixed refresh on changes bugs

kobizz committed:
Merge branch 'release/2.4.2' of https://github.com/pojome/elementor into release/2.4.2

nuritsha committed:
core/upgrade: fix - delete elementor_log from options table for versions before 2.4.2

nuritsha committed:
core/upgrade: fix - limit to free only

nuritsha committed:
core/upgrade: fix - limit to free only

KingYes committed:
Merge pull request #6800 from nuritsha/fix/remove-elementor-log

core/upgrade: fix - delete elementor_log from options table for versions before 2.4.2

kobizz committed:
Removed `revisions` key from testing

kobizz committed:
Merge branch 'release/2.4.2' of https://github.com/pojome/elementor into release/2.4.2

kobizz committed:
Using variable

KingYes committed:
Merge pull request #6799 from kobizz/release/2.4.2

Release/2.4.2

KingYes committed:
Merge pull request #6794 from ramiy/patch-2

i18n: use %s placeholder in translation strings

KingYes committed:
Merge pull request #6795 from ramiy/patch-3

i18n: use %s placeholder in translation strings

KingYes committed:
Merge pull request #6796 from ramiy/patch-4

i18n: use %s placeholder in translation strings

KingYes committed:
Merge pull request #6791 from ramiy/patch-6

Bug Fix: styling image widget captions

kobizz committed:
Page Settings: Update the panel on settings import

kobizz committed:
Choose Control: Update the value also in external change

kobizz committed:
Merge branch 'release/2.4.2' of https://github.com/pojome/elementor into release/2.4.2

KingYes committed:
Merge pull request #6801 from kobizz/release/2.4.2

Release/2.4.2

joshmarom committed:
Adjust j absolutes behaviour to work with new width control

joshmarom committed:
Merge branch 'release/2.4.2' of https://github.com/pojome/elementor into feature/absolute-position
